OpenRA.Mods.Common/Widgets/Logic/Editor/MapSketchImporterLogic.cs

```diff
@@ -0,0 +1,522 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using OpenRA.Widgets;
+
+namespace OpenRA.Mods.Common.Widgets.Logic
+{
+	public class MapSketchImporterLogic : ChromeLogic
+	{
+		Widget panel;
+		DropDownButtonWidget[] terrainDropdowns = new DropDownButtonWidget[13];
+		DropDownButtonWidget[] resourceDropdowns = new DropDownButtonWidget[13];
+		private enum ColorIndex { White, Black, Gray, Red, Orange, Yellow, Green, Teal, LightBlue, DarkBlue, Purple, Pink, Magenta }
+
+		class DropDownOption
+		{
+			public string Title;
+			public Func<bool> IsSelected;
+			public Action OnClick;
+		}
+
+		static Dictionary<int, string> defaultFillerTilesd2k = new Dictionary<int, string>

```

Comment:
Sorry, but you can't hard-code mod specific (d2k) stuff in OpenRA.Mods.**Common**.